Dockerfile

```diff
@@ -1,18 +1,42 @@
 FROM nixos/nix:2.18.8
+
 # default build args
 ARG MAX_JOBS=4
 ARG CORES=4
+
+# Set up Nix configuration and user
 RUN echo "experimental-features = nix-command flakes" >> /etc/nix/nix.conf \
     && echo "max-jobs = $MAX_JOBS" >> /etc/nix/nix.conf \
     && echo "cores = $CORES" >> /etc/nix/nix.conf \
-    && nix profile install nixpkgs#cachix nixpkgs#git-lfs \
-    && cachix use kernel-builder
-WORKDIR /kernelcode
-COPY . /etc/kernel-builder/
+    && echo "trusted-users = root nixuser" >> /etc/nix/nix.conf \
+    # Create user entries directly in password and group files
+    && echo "nixuser:x:1000:1000:NixOS User:/home/nixuser:/bin/bash" >> /etc/passwd \
+    && echo "nixuser:x:1000:" >> /etc/group \
```

Comment:
Any reason not to do `useradd -m -u 1000 -g 1000 nixuser`?

Reply:
I was running into various issues and echo seemed the most consistent in a couple test. Happy to revisit soon but currently this seems to work well (and be relatively readable)